hack/managed-kyma-migration/README.md

```diff
@@ -0,0 +1,103 @@
+# Managed Kyma migration
+
+> **NOTE**: This documentation is relevant for managed Kyma only and does not apply to OS Kyma.
+
+## Scenarios
+
+### Provisioning of API Gateway CR via Lifecycle-Manager in a new cluster
+
+If there is no APIGateway CR, then Lifecycle Manager provisions the default API Gateway CR defined in the API Gateway ModuleTemplate. The migration
+adds the API Gateway module to the Kyma CR.
+
+### Provisioning of Istio CR via Lifecycle Manager in a cluster with existing modules
+
+If there is no APIGateway CR, then Lifecycle Manager provisions the default APIGateway CR defined in the APIGateway ModuleTemplate. The migration
+adds the API Gateway module to the Kyma CR without overwriting existing module configuration.
+
+## Migration test process
+
+### Test scenarios
+
+1. Apply the ModuleTemplate for both `fast` and `regular` channel to DEV Control Plane.
+
+#### SKR without existing modules
+
+1. Create Dev SKR.
+2. Execute the migration.
+3. Verify that `api-gateway-manager` is installed and the APIGateway CR's status is `Ready`.
+4. Verify that `api-gateway` deployment is not present.
+
+#### SKR with an existing module
+
+1. Create Dev SKR.
+2. Add the Keda module to the Kyma CR.
+   ```yaml
+   spec:
+     modules:
+       - name: keda
+   ```
+3. Execute the migration.
+4. Verify that `api-gateway-manager` is installed and the APIGateway CR's status is `Ready`.
+5. Verify that `api-gateway` deployment is not present.
+
+## Module's rollout and migration
+
+### Preparations
+
+1. Executing `kcp taskrun` requires the path to the kubeconfig file of the corresponding Gardener project and permissions to list/get shoots.
+
+### Dev
+
+1. Apply the module template to Dev Control Plane.
```

Comment:
Should we have this manual step here or already do the rollout to the `kyma/module-manifests` internal repository in this step, but don't migrate on stage? Is there some restriction, other than some crazy people could laready enable the module in fast channel on stage? Doing this manual on Dev feels wrong.

Reply:
Yeah, crazy people is the only issue here :D